modules/schedule_module/php/schedulemoduleprovisioner.class.inc

```diff
@@ -0,0 +1,140 @@
+<?php declare(strict_types=1);
+/**
+ * This file implements a data provisioner to the schedule_module menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Core
+ * @package    Main
+ * @subpackage Core
+ * @author     Shen Wang <wangshen.mcin@gmail.com>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\schedule_module;
+require 'data_entry.class.inc';
+require 'DataEntrySelectClause.class.inc';
+require 'MriDataEntryExpression.class.inc';
+require 'BloodCollectionDataEntryExpression.class.inc';
+require 'BehavioralDataEntryExpression.class.inc';
+// require 'MriDataEntryExpression.class.inc';
+/**
+ * This class implements a data provisioner to get all modules
+ * for the schedule_module menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Core
+ * @package    Main
+ * @subpackage Core
+ * @author     Shen Wang <wangshen.mcin@gmail.com>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class ScheduleModuleProvisioner extends \LORIS\Data\Provisioners\DBRowProvisioner
+{
+    /**
+     * Create a ScheduleModuleProvisioner, which gets modules for the
+     * Schedule Module menu table.
+     */
+    function __construct()
+    {
+        parent::__construct(
+            $this->getQuery(),
+            []
+        );
+
+    }
+
+    /**
+     * Returns an instance of a ModuleRow object.
+     *
+     * @param array $row The database row from the LORIS Database class.
+     *
+     * @return \LORIS\Data\DataInstance An instance representing this row.
+     */
+    public function getInstance($row) : \LORIS\Data\DataInstance
+    {
+            return new ModuleRow($row);
+    }
+    /**
+     * Returns a SQL query string.
+     *
+     * @return string A SQL query string.
+     */
+    public function getQuery() : string
+    {
+        $dataEntryColumns          = (new DataEntrySelectClause())
+            ->register("Behavioral", new BehavioralDataEntryExpression())
+            ->register("MRI", new MriDataEntryExpression())
+            ->register("Blood Collection", new BloodCollectionDataEntryExpression())
+            ->getColumns();
+        $dataEntryStatusExpression = "
+    (
+        CASE
+            WHEN NOT (NOW() > StartsAt) THEN
+                'Upcoming'
+            WHEN hasDataEntryComplete THEN
+                IF(
+                    (hasDataEntryInProgress OR hasDataEntryNotStarted),
+                    'In Progress',
+                    'Complete'
+                )
+            WHEN hasDataEntryInProgress THEN
+                'In Progress'
+            WHEN hasDataEntryNotStarted THEN
+                'Not Started'
+            ELSE
+                'No Data Found'
```

Comment:
when I edit the existing appointment type to Blood Collection, I get Data Entry Status 'No Data Found'. is this expected?

Reply:
Yes,it is, if you change the type to MRI, then i will show "In Progress".

Reply:
what's the reason for Blood Collection showing 'No Data Found' instead of 'In Progress'?